src/v/cluster/rm_stm.cc

```diff
@@ -1597,6 +1600,12 @@ ss::future<result<kafka_result>> rm_stm::replicate_seq(
 
     if (session->cache.empty() && session->lock.ready()) {
         _inflight_requests.erase(bid.pid);
+        auto [it, end] = _log_state.idempotent_pids.equal_range(replicate_time);
+        while (it != end) {
+            if (it->second == bid.pid) {
+                _log_state.idempotent_pids.erase(it++);
```

Comment:
it looks like we put a value to idempotent_pids and then immediately remove it once the replication is over, so idempotent_pids is almost always empty